src/v/cluster/topic_recovery_service.cc

```diff
@@ -227,38 +227,15 @@ collect_manifest_paths(
     const auto& bucket = cfg.bucket;
     auto rtc = make_rtc(as, cfg);
 
-    // List only the items at the top of the bucket hierarchy. The delimiter
-    // ensures that any "directories" will be collected into the common_prefixes
-    // field of the result.
-    auto top_level = co_await remote.list_objects(
-      bucket, rtc, std::nullopt, '/');
-    if (top_level.has_error()) {
-        vlog(
-          cst_log.error,
-          "Failed to list top level items: {}",
-          top_level.error());
-        co_return recovery_error_ctx::make("failed to list top level items");
+    // Look under each manifest prefix for topic manifests.
+    std::vector<uint64_t> prefixes;
```

Comment:
If the size is always going to be fixed can an array be used instead of a vector?

Reply:
Done